soc/arm/st_stm32/stm32f3/Kconfig.defconfig.stm32f303xb

```diff
@@ -0,0 +1,14 @@
+# ST Microelectronics STM32F303XB MCU
+
+# Copyright (c) 2023 Martin Gritzan
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_STM32F303XB
+
+config SOC
+	default "stm32f303xb"
```

Comment:
This isn't obvious, but looks like it should be stm32f303xc as STM32F303xB is not an available option here:
https://github.com/zephyrproject-rtos/hal_stm32/blob/d49fb6dbe780b931eee33cce8656e87c46070fa8/stm32cube/stm32f3xx/soc/stm32f3xx.h#L60

So I rather think you should update soc/arm/st_stm32/stm32f3/Kconfig.defconfig.stm32f303xc file

Reply:
Good catch, here's one to the cargo cult 😅 . Now I understand how the HAL target selection comes into place.

I unified both files to a `Kconfig.defconfig.stm32f303x(b-c)`. Inclusion works by globbing and `CONFIG_SOC` is really only used for this purpose [in the HAL's CMakeLists.txt](https://github.com/zephyrproject-rtos/hal_stm32/blob/d49fb6dbe780b931eee33cce8656e87c46070fa8/stm32cube/CMakeLists.txt#L11). I find this looks better. WDYT?